public/app/features/datasources/components/picker/DataSourceList.tsx

```diff
@@ -35,7 +35,7 @@ export interface DataSourceListProps {
 export function DataSourceList(props: DataSourceListProps) {
   const { className, current, onChange } = props;
   // QUESTION: Should we use data from the Redux store as admin DS view does?
-  const dataSources = useGetDatasources({
+  const { dataSources, updateStorage } = useGetDatasources({
```

Comment:
I'm not very convinced about the hook API 🤔 

Just renaming the function, could be already an improvement
```suggestion
  const { dataSources, pushRecentlyUsed } = useGetDatasources({
```

However, this hook is hard to use in other places outside of the picker. Maybe just splitting the hooks in two would leave a single responsibility to each of them:
```js
const { dataSources } = useDatasources(filters)
const { recentlyUsed, pushRecentlyUsedItem } = useRecentlyUsedDatasources();
```
Then, you compose as you need them. No strong opinion. Thoughts @oscarkilhed?

Reply:
Also, thinking in requirements like "move to the top the currently selected element". Probably it makes sense to leave the sorting to the component displaying the list. WDYT?